frontend/src/screens/apps/New.tsx

```diff
@@ -0,0 +1,231 @@
+import { useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useUser } from '../../context/UserContext';
+import { BudgetRenewalType, RequestMethodType, User, UserInfo, nip47MethodDescriptions, nip47MethodIcons, validBudgetRenewals } from '../../types';
+
+const New = () => {
+  const { info } = useUser();
+  // const navigate = useNavigate();
+  const currentUser: User = (info as UserInfo).user!;
+  
+  const location = useLocation();
+  const queryParams = new URLSearchParams(location.search);
+
+  const [edit, setEdit] = useState(false);
+
+  const nameParam = (queryParams.get('name') || queryParams.get('c')) ?? ""
+  const [appName, setAppName] = useState(nameParam);
+  const pubkey = queryParams.get('pubkey') ?? "";
+  const returnTo = queryParams.get('return_to') ?? "";
+
+  const budgetRenewalParam = queryParams.get('budget_renewal') as BudgetRenewalType;
+  const budgetRenewal: BudgetRenewalType = validBudgetRenewals.includes(budgetRenewalParam) ? budgetRenewalParam : "monthly";
+
+  const reqMethodsParam = queryParams.get('request_methods');
+  const [requestMethods, setRequestMethods] = useState(reqMethodsParam ?? Object.keys(nip47MethodDescriptions).join(' '));
+
+  const maxAmountParam = queryParams.get('max_amount') ?? "";
+  const [maxAmount, setMaxAmount] = useState(parseInt(maxAmountParam));
+
+  const parseExpiresParam = (expiresParam: string): string => {
+    if (/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/.test(expiresParam)) {
+      const d = new Date(expiresParam); 
+      const isIso = d instanceof Date && !isNaN(d.getTime()) && d.toISOString() === expiresAtParam;
+      if (isIso) {
+        return expiresParam
+      }
+    }
+    if (!isNaN(parseInt(expiresParam))){
+      return (new Date(parseInt(expiresAtParam as string) * 1000)).toISOString()
+    }
+    return ""
+  }
+
+  // Only timestamp in seconds or ISO string is expected
+  const expiresAtParam = parseExpiresParam(queryParams.get('expires_at') ?? "")
+  const [expiresAt, setExpiresAt] = useState(expiresAtParam ?? "");
+  const [days, setDays] = useState(0);
+  const [expireOptions, setExpireOptions] = useState(false);
+
+  const today = new Date();
+  const handleDays = (days: number) => {
+    setDays(days);
+    if (!days) {
+      setExpiresAt("");
+      return;
+    }
+    const expiryDate = new Date(today.getTime() + days * 24 * 60 * 60 * 1000);
+    setExpiresAt(expiryDate.toISOString());
+  }
+
+  const handleRequestMethodChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const rm = event.target.value;
+    if (requestMethods.includes(rm)) {
+      // If checked and item is already in the list, remove it
+      const newMethods = requestMethods.split(" ").filter((reqMethod) => reqMethod !== rm).join(" ")
```

Comment:
do we need to handle as a string here? what about we use a Set and then only convert it to its final format (join to a string split by spaces) on submission